lib/tracer/tracer.js

```diff
@@ -276,6 +278,62 @@ Tracer.prototype.buildTrace = function (traceInfo) {
   return traceInfo;
 };
 
+/**
+ * There are two formats for traces. While the method/publication is running, the trace is in the object format.
+ * This is easier to work with, but takes more space to store. After the trace is complete (either finished or errored),
+ * it is built which among other things converts the events to the array format.
+ *
+ * The key difference of `optimizeEvent` and `optimizeEvents` is that they do not mutate the original events.
+ */
+Tracer.prototype.optimizeEvent = function (objectEvent) {
+  const {at, endAt, stack, nested = [], forcedEnd, name} = objectEvent;
+
+  const computeTime = at && endAt ? endAt - at : 0;
+
+  const optimizedNestedEvents = nested.length ? this.optimizeEvents(nested) : undefined;
```

Comment:
This can use `this._hasUsefulNested` to check when it should keep the nested events. Otherwise, there ends up being a lot of extra async events which isn't useful and makes the traces larger.
https://github.com/monti-apm/monti-apm-agent/blob/a114f815a0639f3cdd14eaee6487a2c3b4eac147/lib/tracer/tracer.js#L174